go/arrow/array/fixedsize_binarybuilder.go

```diff
@@ -81,6 +83,12 @@ func (b *FixedSizeBinaryBuilder) AppendNull() {
 	b.UnsafeAppendBoolToBitmap(false)
 }
 
+func (b *FixedSizeBinaryBuilder) AppendEmptyValue() {
+	b.Reserve(1)
+	b.values.Advance(b.dtype.ByteWidth)
```

Comment:
@pitrou There's no undefined values in Go, everything is defaulted to a `zero` value based on the type. For `byte` it's 0. So by just advancing the underlying buffer it automatically zeroes out the memory. Since this is for FixedSizeBinary, it needs to get advanced like this (the C++ version of this just explicitly zeroes out the memory which, as I said, isn't needed in Go)